src/contexts/walletconnect.tsx

```diff
@@ -204,24 +264,18 @@ export const WalletConnectContextProvider: FC = ({ children }) => {
             const result = await client.web3.request(p)
             await walletConnectClient.respond({
               topic: event.topic,
-              response: {
-                id: event.id,
-                jsonrpc: '2.0',
-                result
-              }
+              response: { id: event.id, jsonrpc: '2.0', result }
             })
             break
           }
           case 'alph_requestExplorerApi': {
             const p = request.params as ApiRequestArguments
-            const result = await client.explorer.request(p)
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const call = (client.explorer as any)[`${p.path}`][`${p.method}`] as (...arg0: any[]) => Promise<any>
+            const result = await call(...p.params)
```

Comment:
I am not sure why this change was needed, maybe @Lbqds has more info as to why `client.explorer.request(p)` was not sufficient.

Reply:
The dApp connected through WalletConnect converts the request parameters to the `ApiRequestArguments` type in `@alephium/web3` (code is available [here](https://github.com/alephium/alephium-web3/blob/18b7c2b0a457144367bd12cac94ed46b4ae24419/packages/web3/src/api/types.ts#L251)) when sending a request. However, the `ExplorerClient` used by desktop-wallet comes from `@alephium/sdk`, which cannot directly handle requests of type `ApiRequestArguments`. Therefore, we need to make this change, and the code for `@alephium/web3` is available [here](https://github.com/alephium/alephium-web3/blob/18b7c2b0a457144367bd12cac94ed46b4ae24419/packages/web3/src/api/types.ts#L262).